### PR TITLE
build: Update dependencies on `@fluidframework/build-common` to `2.0.1`

### PR DESCRIPTION
**Pull request by Josmithr**
Updates all deps across the repo.

Also fixes outdated formatting in a TypeScript module in `telemetry-generator`.

### CODE REVIEW

File: azure/packages/azure-client/package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
 		"@fluidframework/aqueduct": "workspace:~",
 		"@fluidframework/azure-client-previous": "npm:@fluidframework/azure-client@2.0.0-internal.7.0.0",
 		"@fluidframework/azure-local-service": "workspace:~",
-		"@fluidframework/build-common": "^2.0.0",
+		"@fluidframework/build-common": "2.0.1",
```
Comment (Josmithr) [inline azure/packages/azure-client/package.json:64]: I missed the `^` in the updates here. Will fix.

**Comment by tylerbutler** [comment]
Did you do this with flub bump deps or through some other means?

**Comment by Josmithr** [comment]
> Did you do this with flub bump deps or through some other means?

Find and replace. I forgot that was something I could do with flub.

**Comment by tylerbutler** [comment]
> Is there anything additional that the flub command would have done here? I can regen the changes if needbe.

No, I don't think so. It would have run pnpm i for you in all the affected places, which may have saved you some time, but nothing other than that. However, were you upgrading a release group dependency, using flub ensures it upgrades all the packages in that release group. There was a recent half-upgrade of build-tools that would have been avoided by using flub. Oh, and it would have done a nice commit with a nice summary of the changes for the PR description.

That said, my question was not a leading one - I was honestly just curious as feedback for the build-tools in general. Upgrading deps is something that there's no "wrong way," really, unless we define policies and enforce them. And we sort of already do that as best we can with tools and CI checks, so your approach was definitely not wrong.

**Comment by Josmithr** [comment]
> > Is there anything additional that the flub command would have done here? I can regen the changes if needbe.
> 
> No, I don't think so. It would have run pnpm i for you in all the affected places, which may have saved you some time, but nothing other than that. However, were you upgrading a release group dependency, using flub ensures it upgrades all the packages in that release group. There was a recent half-upgrade of build-tools that would have been avoided by using flub. Oh, and it would have done a nice commit with a nice summary of the changes for the PR description.
> 
> That said, my question was not a leading one - I was honestly just curious as feedback for the build-tools in general. Upgrading deps is something that there's no "wrong way," really, unless we define policies and enforce them. And we sort of already do that as best we can with tools and CI checks, so your approach was definitely not wrong.

Just wanted to double check. There might be some things we can do to enhance discoverability a bit. When I started updating, I briefly checked `build-cli/README.md` to double check if this was supported. I parsed the brief description of the `bump` command there to be a more wholistic update step, and erroneously concluded that the command was strictly related to bumping local package versions (and then fixing up local deps to match), not that it could do two separate things (bump local versions **or** bump deps, even of external things). So maybe a phrasing update there would be helpful.

Otherwise, maybe adding a "Working in the repo" section to our root README with recommended and/or example patterns for accomplishing things like this would be good to enhance discovery (that is always where I start when looking for guides).

**Comment by Josmithr** [comment]
> > Otherwise, maybe adding a "Working in the repo" section to our root README with recommended and/or example patterns for accomplishing things like this would be good to enhance discovery (that is always where I start when looking for guides).
> 
> There's this in the wiki: https://github.com/microsoft/FluidFramework/wiki/Managing-dependencies Maybe we should link to it in the readmes?

Yeah, I wasn't aware of that doc. It would definitely make it more discoverable for how I generally approach things if it were linked from the root README.

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**